tracee-ebpf/tracee/events_pipeline.go

```diff
@@ -38,6 +45,140 @@ func (t *Tracee) handleEvents(ctx gocontext.Context) {
 	t.WaitForPipeline(errcList...)
 }
 
+// Under some circumstances, tracee-rules might be slower to consume events
+// than tracee-ebpf is capable of generating them. This requires
+// tracee-ebpf to deal with this possible lag, but, at the same,
+// perf-buffer consumption can't be left behind (or important events coming
+// from the kernel might be loss, causing detection misses).
+//
+// There are 3 variables connected to this issue:
+//
+// 1) perf buffer could be increased to hold very big amount of memory
+//    pages: The problem with this approach is that the requested space,
+//    to perf-buffer, through libbpf, has to be contiguous and it is almost
+//    impossible to get very big contiguous allocations through mmap after
+//    a node is running for some time.
+//
+// 2) raising the events channel buffer to hold a very big amount of
+//    events: The problem with this approach is that the overhead of
+//    dealing with that amount of buffers, in a golang channel, causes
+//    event losses as well. It means this is not enough to relief the
+//    pressure from kernel events into perf-buffer.
+//
+// 3) create an internal, to tracee-ebpf, buffer based on the node size.
+
+// queueEvents implements an internal FIFO queue for caching events
+func (t *Tracee) queueEvents(ctx gocontext.Context, in <-chan *external.Event) (chan *external.Event, chan error) {
```

Comment:
As mentioned in my comment, maybe we could use a Queue interface here instead of hardcoding this mechanism?